modules/image-loading-optimization/assets/js/editor/hooks/check-lcp-image-before-save.js

```diff
@@ -0,0 +1,138 @@
+
+        if( lcpImages.length ) {
+            const imgElements = this.getImageElementsList( elementor.getPreviewContainer().children );
+            this.setLoadingOptimizationSettings( imgElements, lcpImages );
+        }
+
+        this.switchDeviceMode( currentDeviceMode );
+        this.getPreviewFrame().scrollTo( 0, currentScrollPosition );
+
+        return true;
+    }
+
+    switchDeviceMode( newDeviceMode, currentDeviceMode = null ) {
+        if ( ! currentDeviceMode ) {
+            currentDeviceMode = this.getCurrentDeviceMode();
+        }
+
+        if( currentDeviceMode !== newDeviceMode ) {
+            $e.run( 'panel/change-device-mode', { device: newDeviceMode } );
+        }
+    }
+
+    async getWebVitalsFromPreview( device ) {
+        return await new Promise( (resolve, reject) => {
+            // Listen for message from preview frame.
+            let eventRef = window.addEventListener("message", (event) => {
+                if (event.data?.name && event.data?.name == 'elementor/web-vitals/editor') {
+                    window.removeEventListener("message", eventRef);
+                    resolve(event.data);
+                }
+            });
+            
+            // Send message to preview frame to request web vitals.
+            this.getPreviewFrame().postMessage(
+                { 
+                    device,
+                    name: 'elementor/web-vitals/preview',
+                },
+                '*',
+            );
+        });
+    }
+
+    getImageElementsList( elements ) {
+        let flatted = [];
+        for ( const element of elements ) {
+            if( element?.label === 'Image' ){
+                flatted.push( element );
+            }
+            if ( element.children.length ) {
+                flatted = flatted.concat( this.getImageElementsList( element.children ) );
+            }
+        }
+        return flatted;
+    }
+
+    setLoadingOptimizationSettings( elements, lcpImages ) {
+        elements.forEach( ( element ) => {
+            // Use the internal command because it's not a user action
+            $e.internal( 'document/elements/set-settings', {
+                container: element,
+                settings: {
+                    is_lcp_image: this.isLCPImage( element, lcpImages ), // Todo: if `false` then remove the setting
+                    is_above_the_fold: this.isElementAboveTheFold( element ),
+                },
+                options: {
+                    external: true, // Update panel if it's open
+                    render: false, // Don't re-render the element
+                },
+            } );
+        } );
+    }
+
+    isLCPImage( element, lcpImages ) { 
+        for ( const lcpImage of lcpImages ) {
+            if( lcpImage.widgetId === element.id ) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    isElementAboveTheFold( element ) {
```

Comment:
This is currently above the fold for desktop mode only. If we make device specific we need to figure that out.